src/adapter/src/coord/timestamp_selection.rs

```diff
@@ -161,14 +158,22 @@ impl<S: Append + 'static> Coordinator<S> {
             candidate.advance_by(since.borrow());
         }
 
-        if use_timestamp_oracle {
-            let timeline = timeline
-                .as_ref()
-                .expect("timeline is always present when using the timestamp oracle");
-            let timestamp_oracle = self.get_timestamp_oracle(timeline);
-            oracle_read_ts = Some(timestamp_oracle.read_ts());
-            candidate.join_assign(&oracle_read_ts.unwrap());
-        } else if when.advance_to_upper() {
+        if let Some(timeline) = &timeline {
+            if (isolation_level == &vars::IsolationLevel::StrictSerializable
+                && when.can_advance_to_global_ts())
+                || when.must_advance_to_global_ts()
+            {
+                let timestamp_oracle = self.get_timestamp_oracle(timeline);
```

Comment:
The functions have `global_ts` in their names, but this uses the timeline's oracle. Can the names be changed to accurately describe the requirements? Or maybe code needs to be added somewhere that only does this if the timeline is epochms?

Reply:
Good point, I've changed the name of these functions.